diff -rupN freetype-2.4.8.old//include/freetype/config/ftoption.h freetype-2.4.8/include/freetype/config/ftoption.h
--- freetype-2.4.8.old//include/freetype/config/ftoption.h	2012-02-19 10:44:01.307000022 -0800
+++ freetype-2.4.8/include/freetype/config/ftoption.h	2012-02-19 10:45:32.002000021 -0800
@@ -590,7 +590,7 @@ FT_BEGIN_HEADER
   /*   This option requires TT_CONFIG_OPTION_BYTECODE_INTERPRETER to be    */
   /*   defined.                                                            */
   /*                                                                       */
-/* #define TT_CONFIG_OPTION_SUBPIXEL_HINTING */
+#define TT_CONFIG_OPTION_SUBPIXEL_HINTING
 
 
   /*************************************************************************/
@@ -610,7 +610,7 @@ FT_BEGIN_HEADER
   /*   This option requires TT_CONFIG_OPTION_SUBPIXEL_HINTING to be        */
   /*   defined.                                                            */
   /*                                                                       */
-/* #define TT_CONFIG_OPTION_SUBPIXEL_HINTING_ADDITIONAL_TWEAKS */
+#define TT_CONFIG_OPTION_SUBPIXEL_HINTING_ADDITIONAL_TWEAKS
 
 
   /*************************************************************************/
