diff -rupN freetype-2.4.8.old//include/freetype/config/ftoption.h freetype-2.4.8//include/freetype/config/ftoption.h
--- freetype-2.4.8.old//include/freetype/config/ftoption.h	2012-02-19 10:33:42.195000020 -0800
+++ freetype-2.4.8//include/freetype/config/ftoption.h	2012-02-19 10:34:32.771000523 -0800
@@ -92,7 +92,7 @@ FT_BEGIN_HEADER
   /* This is done to allow FreeType clients to run unmodified, forcing     */
   /* them to display normal gray-level anti-aliased glyphs.                */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
@@ -761,7 +761,7 @@ FT_BEGIN_HEADER
    * is recommended to disable the macro since it reduces the library's code
    * size and activates a few memory-saving optimizations as well.
    */
-#define FT_CONFIG_OPTION_OLD_INTERNALS
+/* #define FT_CONFIG_OPTION_OLD_INTERNALS */
 
 
   /*
