From http://www.infinality.net/forum/viewtopic.php?f=2&t=55

diff --git a/src/truetype/ttgload.c b/src/truetype/ttgload.c
--- a/src/truetype/ttgload.c
+++ b/src/truetype/ttgload.c
@@ -1930,6 +1930,7 @@
         grayscale = grayscale_hinting = TRUE;
         subpixel_hinting = FALSE;
       }
+      if ( FT_IS_TRICKY( glyph->face ) ) { subpixel_hinting = grayscale_hinting = FALSE; }

       exec->enhanced = ( subpixel_hinting
                       || grayscale_hinting );
